pkgs/applications/misc/tiramisu/default.nix

```
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "tiramisu";
-  version = "2.0.20211107";
+  version = "unstable-2023-03-29";
```

```suggestion
  version = "2.0.20211107-unstable-2023-03-29";
```

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

---

"2.0.20211107-unstable-2023-03-29" seems kind of confusing with two dates and all.
Will "2.0-unstable-2023-03-29" suffice?

---

yeah, it does look a bit "sub-optimal"

but here `2.0.20211107` refers to a tag.

maybe others can chime in ?

---

As the author of #234201, @7c6f434c, may I ask you for help?

---

Technically, upstream has 2.0 and _two_ 2.0.X releases based on datestamps. So if we want both to have an unstable word-mark and proper version sorting in case of a plausible 2.0.2024MMDD upstream tag, we need to do this not so nicely looking version with two dates. This is not a question of wording in the documentation, this is a logical implication.

But then the documentation says «should» not «must» for a good reason. So something in the direction of 2.0.20240101-unstable with a comment in the expression would not be completely unreasonable, if people involved with the package agree.

---

I see. So will changing the version to 2.0.20240408-unstable (today's date) suffice?

---

@JohnRTitor I think we are arriving at `2.0.20240408-unstable`, except I would probably use date of commit not today's date.

@7c6f434c re wording in documentation. it also says `The date must be in "YYYY-MM-DD" format."`

so maybe an other option is `2.0-unstable-YYYY-MM-DD`. whichever option we choose we will need to leave a comment

---

Something along the lines of `2.0.20240408-unstable` would be more consistent with the upstream versioning scheme.
2.0-unstable-YYYY-MM-DD is also a viable option that is consistent with Nix versioning scheme.

---

I would say that 2.0.20240408-unstable is breaking a «should» and representing it as «_faking_ the upstream tag in _their_ date format» not «creating a downstream tag based on upstream tag and date». 

2.0-unstable-X has the ordering drawbacks, although it might be irrelevant for the specific package.

---

Changed to `2.0.20240101-unstable`.

---

Can this be resolved now?

---

No, certainly not `20240101`.

Both `2.0-unstable-2023-03-29` and `2.0.20211107-unstable-2023-03-29` should be OK, choose the one you prefer.

---

Apologies for misunderstanding. I decided to go with `2.0-unstable-2023-03-29`.